src/Workspaces/Core/Portable/Shared/Utilities/RoslynParallel.NetFramework.cs

```diff
@@ -0,0 +1,652 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !NET
+
+#pragma warning disable CA1068 // CancellationToken parameters must come last
+#pragma warning disable IDE0007 // Use implicit type
+#pragma warning disable IDE2003 // Blank line required between block and subsequent statement
+#pragma warning disable IDE2004 // Blank line not allowed after constructor initializer colon
+#pragma warning disable VSTHRD200 // Use "Async" suffix for async methods
+
+// Ported from
+// https://github.com/dotnet/runtime/blob/main/src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/Parallel.ForEachAsync.cs
```

Comment:
to be very clear.  this is effectively the runtime's code.  just tweaked as *little* as possible to get working on netfx.

i do *not* want to change it beyond that, as that will be much harder to keep in sync and validate against the netcore impl.